### DISCUSSION TITLE
morph markers not present when using artisan view:cache

### DISCUSSION THREAD

**Discussion by lgebing** [livewire/livewire#7526] [7. Bugs (archived)] [answered]
### Livewire version

v3.3.0

### Laravel version

10.37.1

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Create a simple view containing a conditional, for example:
```
<div>
    @if (true)
        Hello world
    @endif
</div>
```
2. Inspect the element in the browser to verify the presence of the morph markers:
![Bildschirmfoto 2023-12-13 um 13 30 38](https://github.com/livewire/livewire/assets/60090387/dd3fce67-cac2-4fe6-81bf-57384b17aa2c)

3. Run `php artisan view:cache`.

4. Reload the page and inspect the element in the browser to see that it does not contain morph markers:
![Bildschirmfoto 2023-12-13 um 13 32 19](https://github.com/livewire/livewire/assets/60090387/9a8fdbce-be42-442c-8255-1f8bddbc2b2d)

5. (Optional) Run `php artisan view:clear`, reload the page and check the element again. The morph markers will now be present again.

### Repository or Wirebox URL

https://wirebox.app/b/xp31q

### Additional information

The Wirebox provided should contain the code necessary to reproduce the issue (minus the artisan commands), but I could not test it since the preview would not load. Thats why the "Steps To Reproduce" contains slighty different code samples.

Regarding discussions https://github.com/livewire/livewire/discussions/5658 was the closest equivalent I could find, but since it is a slighty different issue (although the cause might be the same) I decided to open a new discussion.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by huubvdw** [comment]
+1 on this issue. After we migrated from v2 to v3, this led to numerous errors `Uncaught (in promise) Component not found ...` in our application on production (where view cache is enabled).
This is a serieus concern, since our developers now need to run `view:cache` after every change to verify everything works as expected.

**Comment by jplhomer** [comment]
I would LOVE to help get this fixed! Upgrading a legacy app from v2 to v3 leads to a ton of errors which are hard to catch and debug. I've disabled the morph markers for now to match development and production environments, but it's just brutal trying to figure out what's wrong and how Livewire/Alpine's diffing engine works.

Properly injecting morph markers would be so great in this case.

Anybody know where to start contributing a fix for this?

**Reply by joshhanley** [reply]
@jplhomer I suspect the issue is that Livewire injects the morphmarkers only when a Livewire component view is being parsed. This is because the component class has booted before it, which would trigger the morphmarkers logic to enable the injections to happen.

The problem is that I'm guessing `view:cache` just iterates through the views directory and uses blade's standard engine to parse them. But that means Livewire views aren't being parsed correctly.

I'm not sure how to even get that to work (or if it's even possible). I suspect it would mean all blade views would need to be rendered with morph markers injected (some people may not like that).

**Reply by thumbtech** [reply]
@joshhanley if using

`'inject_morph_markers' => false,`

does this bug still exist?

I assume using morph markers and not using view:cache is a net positive from a performance standpoint?

**Reply by joshhanley** [reply, marked as answer]
@thumbtech the bug still exists, but we have a working solution in PR #8068, just waiting for Caleb to review it.
